sql3/test/defs/defs_groupby.go

```diff
@@ -257,3 +257,171 @@ var groupByTests = TableTest{
 		},
 	},
 }
+
+// groupby/distinct with sets tests
+var groupBySetDistinctTests = TableTest{
+	Table: tbl(
+		"groupby_set_test",
+		srcHdrs(
+			srcHdr("_id", fldTypeID),
+			srcHdr("ids1", fldTypeIDSet),
+			srcHdr("ss1", fldTypeStringSet),
+		),
+		srcRows(
+			srcRow(int64(1), []int64{1, 2}, []string{"a", "b"}),
+			srcRow(int64(2), []int64{3, 4}, []string{"d", "e"}),
+			srcRow(int64(3), []int64{1, 4}, []string{"a", "d"}),
+			srcRow(int64(4), []int64{3, 2}, []string{"c", "b"}),
+		),
+	),
+	SQLTests: []SQLTest{
+		{
+			SQLs: sqls(
+				"select distinct ids1 from groupby_set_test with (flatter(foo))",
+			),
+			ExpErr: "unknown query hint 'flatter'",
```

Comment:
I was already moving the mouse to say "i think i found a typo" when my reading caught up with the expected error. :)